osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -0,0 +1,401 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.Collections.Generic;
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Configuration;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Transformations;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Backgrounds;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Modes;
+using osu.Framework.Allocation;
+using osu.Framework.Input;
+using OpenTK.Input;
+using System.Linq;
+
+namespace osu.Game.Overlays.Mods
+{
+    public class ModSelectOverlay : WaveOverlayContainer
+    {
+        private const int button_duration = 700;
+        private const int ranked_multiplier_duration = 700;
+        private const float content_width = 0.8f;
+
+        private Color4 lowMultiplierColour, highMultiplierColour;
+
+        private OsuSpriteText rankedLabel, multiplierLabel;
+        private FlowContainer rankedMultiplerContainer;
+
+        private FlowContainer modSectionsContainer;
+
+        private DifficultyReductionSection difficultyReductionSection;
+        private DifficultyIncreaseSection difficultyIncreaseSection;
+        private AssistedSection assistedSection;
+        private ModSection[] sections

```

Comment:
Creating a new array with every get seems weird. Either make the array a member and return it or make this a method `CreateSectionsArray`.